src/main/java/com/github/cameltooling/lsp/internal/parser/CamelEIPParser.java

```diff
@@ -0,0 +1,51 @@
+package com.github.cameltooling.lsp.internal.parser;
+
+import org.eclipse.lsp4j.CompletionItem;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.TextDocumentItem;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.regex.Pattern;
+
+public class CamelEIPParser {
+
+	private final TextDocumentItem document;
+
+	public CamelEIPParser(TextDocumentItem textDocumentItem) {
+		this.document = textDocumentItem;
+	}
+
+	public boolean canPutEIP(Position position) {
+		Pattern CHOICE_EIP_PATTERN = Pattern.compile("\\)\\s*[.[c[h[o[i[c[e]?]?]?]?]?]?]?\\(?$", Pattern.MULTILINE);
```

Comment:
This is ugly :D, but I didn't feel like doing a Trie based implementation was justified with just one EIP, which was my original goal

I'll be refactoring this when I add more EIPs